src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -436,21 +437,21 @@ def __init__(
             resend_latest: Whether to resend latest channel values to newly created
                 receivers, like in `Broadcast` channels.
         """
-        self._engine: _GenericEngine = engine
+        self._engine = engine
         super().__init__(name, resend_latest)
 
     @property
-    def engine(self) -> _GenericEngine:
+    def engine(self) -> FormulaEngine | FormulaEngine3Phase:
         """Return the formula engine attached to the channel.
 
         Returns:
             A FormulaEngine instance.
         """
         return self._engine
 
-    def new_receiver(
+    def new_receiver(  # type: ignore[override]
```

Comment:
Because `_BaseFormulaChannel` inherits `Broadcast`,  we should inherit the type of its methods as well,  i.e. this `new_receiver` should return `Receiver[Sample]`, etc.  But we are returning `FormulaReceiver` which derives from `Receiver[Sample]`,  but is still a different type with additional methods.  and mypy doesn't like this.

Reply:
This is no longer relevant,  because `FormulaReceiver` are gone now.  FormulaEngines directly return broadcast receivers.